airflow/providers/slack/transfers/sql_to_slack.py

```diff
@@ -165,3 +193,115 @@ def execute(self, context: Context) -> None:
         self._render_and_send_slack_message(context, df)
 
         self.log.debug('Finished sending SQL data to Slack')
+
+
+class SqlToSlackApiFileOperator(BaseSqlToSlackOperator):
```

Comment:
What are our options if we want to avoid creating a new operator for files?

Reply:
Wow! . Quick question then - I am  preparing for Provider's release. SHould I merge this one (code looks cool but I guess I need TL;DR; if the current state in this PR is "Releasable" if we merge it). I guess so, but wanted to get the feeling of others involved in the disucssion :)

Reply:
I dont think we should add operator just to send file. I understand the challange but I think we should try to mitigate this in the operator itself.

If Slack offers 3 diffrent approches than maybe we should have base class and 3 operators one per approch? Then each operator will be able to levrage the full capabilities of what slack offers?

Reply:
I think we do not need to rush especially if we do not have agreement how to do it in a better way.

IMHO. For send SQL response as a message we could actually do by three different way without turn into the pain 🤣 
most of the major parameters are presented in all 3 APIs requests.

For files situation are different we can use only Slack API and internally it do not have same parameters from different methods. Most close it `text` from `chat.postMessage` and `initial_comment` from `files.upload` even `channel` and `channels` working differently.

Reply:
@Taragolis Under the assumption that we do not want to add a new operator for this - what are our options?
I think we are having hard time here since the capabilities are not clear on the hook level.
If slack has 3 different APIs to send message maybe we should have 3 hooks ?

Reply:
Initially when I started refactor slack provider I've also want to create additional hook but after deeply investigate I've found that Legacy and based on slack API Incoming Webhook has almost the same interface.
Legacy supports additional features which is not available in new one: change icons, username and channels. 

And it is impossible (just my personal findings) to determine witch Webhook URL use for Legacy Incoming Webhook and which one for new one - even HTTP responses almost the same. That is why I combine usage in `airflow.providers.slack.hooks.slack_webhook.SlackWebhookHook` with warnings about parameters which supported by Legacy only

If we want to just **send message as a text** than we might create some of generic interface because all three methods support [blocks](https://api.slack.com/block-kit) and fallback messages.

With send as file (upload) it is quite difficult because it is different method of API with different parameters and it only supported by Slack API not Webhook. And in case of send as file we do not need overwrite `render_template_fields` for add Jinja filter in runtime